src/Components/WebView/WebView/src/WebViewManager.cs

```diff
@@ -204,25 +206,29 @@ record RootComponent
         /// Disposes the current <see cref="WebViewManager"/> instance.
         /// </summary>
         /// <param name="disposing"><c>true</c> when dispose was called explicitly; <c>false</c> when it is called as part of the finalizer.</param>
-        protected virtual void Dispose(bool disposing)
+        protected virtual async ValueTask DisposeAsync(bool disposing)
         {
             if (!_disposed)
             {
                 if (disposing)
                 {
-                    _currentPageContext?.Dispose();
+                    if (_currentPageContext != null)
+                    {
+                        await _currentPageContext.DisposeAsync();
+                    }
                 }
 
                 _disposed = true;
             }
         }
 
         /// <inheritdoc/>
-        public void Dispose()
+        public ValueTask DisposeAsync()
         {
-            // Do not change this code. Put cleanup code in 'Dispose(bool disposing)' method
-            Dispose(disposing: true);
+            // Do not change this code. Put cleanup code in 'DisposeAsync(bool disposing)' method
+            var task = DisposeAsync(disposing: true);
             GC.SuppressFinalize(this);
```

Comment:
Nit: I'd just suppress finalization first. I don't see how order could matter here. And it's a bit weird to do it potentially in parallel with DisposeAsync.

Reply:
See other comment; this is roughly what's in the docs.

Reply:
We don't need to do the whole pattern here since this class (nor derived classes) have a finalizer, am I wrong?

Reply:
Removed 'bool disposing'